### PR TITLE
store: fix potential panic in GC worker

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
Try to fix
```
goroutine 5109074295 [running]:
github.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1(0xc01b732830, 0x23b8c60, 0xc000138000, 0xc7ef2ca7e0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/adapter.go:229 +0x48c
panic(0x1ebe500, 0x31c4a60)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb/session.(*TxnState).LockKeys(0xc2e81aa1f0, 0x23b8c60, 0xc000138000, 0xc03018ab30, 0x5bd729875b001b6, 0xc350, 0xc817a17540, 0x1, 0x1, 0xc0638be8c0, ...)
	<autogenerated>:1 +0x36
github.com/pingcap/tidb/executor.doLockKeys(0x23b8c60, 0xc000138000, 0x23effa0, 0xc2e81aa1e0, 0xc350, 0xc817a17540, 0x1, 0x1, 0x13, 0xc2bdd5e300)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/executor.go:800 +0x185
github.com/pingcap/tidb/executor.(*SelectLockExec).Next(0xc33d69de60, 0x23b8c60, 0xc000138000, 0xca6dc03680, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/executor.go:789 +0x5a8
github.com/pingcap/tidb/executor.Next(0x23b8c60, 0xc000138000, 0x23bf820, 0xc33d69de60, 0xca6dc03680, 0x0, 0x203000)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/executor.go:192 +0xbd
github.com/pingcap/tidb/executor.(*ProjectionExec).unParallelExecute(0xc0148a7100, 0x23b8c60, 0xc000138000, 0xca27605ce0, 0xc3a2a6, 0xc013663180)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/projection.go:182 +0x8d
github.com/pingcap/tidb/executor.(*ProjectionExec).Next(0xc0148a7100, 0x23b8c60, 0xc000138000, 0xca27605ce0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/projection.go:169 +0xf4
github.com/pingcap/tidb/executor.Next(0x23b8c60, 0xc000138000, 0x23bf660, 0xc0148a7100, 0xca27605ce0, 0xca2698aa98, 0xc00b0e9080)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/executor.go:192 +0xbd
github.com/pingcap/tidb/executor.(*recordSet).Next(0xc01b732578, 0x23b8c60, 0xc000138000, 0xca27605ce0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/adapter.go:113 +0xc7
github.com/pingcap/tidb/executor.(*ExecStmt).runPessimisticSelectForUpdate(0xc7ef2ca7e0, 0x23b8c60, 0xc000138000, 0x23bf660, 0xc0148a7100, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/adapter.go:387 +0x303
github.com/pingcap/tidb/executor.(*ExecStmt).handlePessimisticSelectForUpdate(0xc7ef2ca7e0, 0x23b8c60, 0xc000138000, 0x23bf660, 0xc0148a7100, 0x355a9c0, 0xc1a4df2b00, 0x0, 0xc03018a800)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/adapter.go:362 +0x5d
github.com/pingcap/tidb/executor.(*ExecStmt).Exec(0xc7ef2ca7e0, 0x23b8c60, 0xc000138000, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/executor/adapter.go:287 +0x70c
github.com/pingcap/tidb/session.runStmt(0x23b8c60, 0xc000138000, 0x23effa0, 0xc2e81aa1e0, 0x23bf1e0, 0xc7ef2ca7e0, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/session/tidb.go:249 +0x1cb
github.com/pingcap/tidb/session.(*session).executeStatement(0xc2e81aa1e0, 0x23b8c60, 0xc000138000, 0x0, 0x23beba0, 0xc0183f58c0, 0x23bf1e0, 0xc7ef2ca7e0, 0x0, 0x0, ...)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/session/session.go:976 +0x1ce
github.com/pingcap/tidb/session.(*session).execute(0xc2e81aa1e0, 0x23b8c60, 0xc000138000, 0xc1a4df2b60, 0x63, 0xc9c65c, 0x31c8260, 0xc0176dc880, 0x11, 0xc01b733190)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/session/session.go:1091 +0x96a
github.com/pingcap/tidb/session.(*session).Execute(0xc2e81aa1e0, 0x23b8c60, 0xc000138000, 0xc1a4df2b60, 0x63, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/session/session.go:1014 +0xdb
github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).loadValueFromSysTable(0xc002626200, 0x2104528, 0xc, 0x0, 0x0, 0x0, 0x0)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:1274 +0x119
github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).checkUseDistributedGC(0xc002626200, 0x23b8c20, 0xc003ae2400, 0x5bd722345b00000)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:749 +0x55
github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).runGCJob(0xc002626200, 0x23b8c20, 0xc003ae2400, 0x5bd722345b00000, 0xc)
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:534 +0x1451
created by github.com/pingcap/tidb/store/tikv/gcworker.(*GCWorker).leaderTick
	/home/jenkins/agent/workspace/tidb_release-3.0/go/src/github.com/pingcap/tidb/store/tikv/gcworker/gc_worker.go:281 +0xd96
```

### What is changed and how it works?
Do not create a private session for `checkLeader`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -1067,42 +1067,39 @@ func (w *GCWorker) doGC(ctx context.Context, safePoint uint64, concurrency int)
 
 func (w *GCWorker) checkLeader() (bool, error) {
 	metrics.GCWorkerCounter.WithLabelValues("check_leader").Inc()
-	se := createSession(w.store)
-	defer se.Close()
```
Comment (lysu) [inline store/tikv/gcworker/gc_worker.go]: @hicqu  https://github.com/pingcap/tidb/pull/4453 PTAL

File: store/tikv/gcworker/gc_worker.go [hunk 2]
```diff
@@ -155,7 +153,8 @@ func (w *GCWorker) start(ctx context.Context, wg *sync.WaitGroup) {
 	logutil.Logger(ctx).Info("[gc worker] start",
 		zap.String("uuid", w.uuid))
 
-	w.session = createSession(w.store)
+	se := createSession(w.store)
+	defer se.Close()
```
Comment (MyonKeminta) [inline store/tikv/gcworker/gc_worker.go]: It seems the session is not used in this function

Comment (jackysp) [reply store/tikv/gcworker/gc_worker.go]: addressed

**Comment by coocood** [comment]
LGTM

**Comment by bb7133** [comment]
Hi @jackysp , please update the release note in the PR description, thanks!

**Pull request merged**